### PR TITLE
Wicket 6662

### PR DESCRIPTION
**Pull request by bitstorm**
see discussion at https://issues.apache.org/jira/browse/WICKET-6662

### CODE REVIEW

**Comment by martin-g** [comment]
I've glanced over several files and the changes look good!
I will make a proper review in the coming days.

But I'd prefer to keep the old classes as deprecated for Wicket 9.x and remove them in Wicket 10.

**Comment by bitstorm** [comment]
@martin-g normally I would do the same thing, but to me these are more classes for internal uses.  I don't think many people use them directly with Wicket. Once we remove them from Wicket code base they are completely useless. But it's not a problem if we decide to keep them in Wicket 9.0.

**Comment by martin-g** [comment]
@bitstorm My reasoning is: If the classes are public then they have to be deprecated before being removed. There is no way to tell how many applications use any of those classes.
There are no any drawbacks in keeping them as deprecated for 1 major release. The size of the jars will be slightly bigger - no one cares about such difference. We won't support those classes since they are deemed to be removed.
We keep them around just for users' convenience.

**Comment by bitstorm** [comment]
@martin-g done! I've reintroduced and deprecated package org.apache.wicket.util.time

File: wicket-core/src/main/java/org/apache/wicket/request/resource/DynamicImageResource.java [hunk 1]
```diff
@@ -19,13 +19,12 @@
 import java.awt.image.BufferedImage;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-
+import java.sql.Time;
```
Comment (martin-g) [inline wicket-core/src/main/java/org/apache/wicket/request/resource/DynamicImageResource.java]: this import looks wrong and unused.

File: wicket-core/src/test/java/org/apache/wicket/ajax/AjaxRequestHandlerTest.java [hunk 2]
```diff
@@ -284,14 +283,14 @@ void ajaxRedirectSetsNoCachingHeaders()
 		tester.startPage(new Wicket3921());
 
 		tester.clickLink("updatePage");
-		assertEquals(Time.START_OF_UNIX_TIME.toRfc1123TimestampString(),
+		assertEquals(Instants.toRFC7231Format(Instant.EPOCH),
```
Comment (martin-g) [inline wicket-core/src/test/java/org/apache/wicket/ajax/AjaxRequestHandlerTest.java]: Why change the RFC ?

File: wicket-core/src/test/java/org/apache/wicket/page/PageAccessSynchronizerTest.java [hunk 3]
```diff
@@ -98,13 +106,13 @@ public void run()
 		T2 t2 = new T2();
 		t2.setName("t2");
 		t1.start();
-		Duration.milliseconds(100).sleep();
+		Thread.sleep(100);
```
Comment (martin-g) [inline wicket-core/src/test/java/org/apache/wicket/page/PageAccessSynchronizerTest.java]: nit: a more direct translation would be: `TimeUnit.MILLISECONDS.sleep(100)`

File: wicket-jmx/src/main/java/org/apache/wicket/jmx/wrapper/RequestCycleSettings.java [hunk 4]
```diff
@@ -117,9 +117,9 @@ public void setResponseRequestEncoding(final String responseRequestEncoding)
 	 * @see org.apache.wicket.jmx.RequestCycleSettingsMBean#setTimeout(java.lang.String)
 	 */
 	@Override
-	public void setTimeout(final String timeout)
+	public void setTimeout(final Duration timeout)
```
Comment (martin-g) [inline wicket-jmx/src/main/java/org/apache/wicket/jmx/wrapper/RequestCycleSettings.java]: I am not sure JMX will properly construct a Duration from the String input.

File: wicket-util/src/main/java/org/apache/wicket/util/string/StringValue.java [hunk 5]
```diff
@@ -548,7 +552,14 @@ public final Double toDoubleObject() throws StringValueConversionException
...
 	public final Duration toDuration() throws StringValueConversionException
 	{
-		return Duration.valueOf(text, locale);
+		try
+		{
+			return Duration.parse(text);
+		}
+		catch (Exception e)
+		{
+			throw new StringValueConversionException("xxx", e);
```
Comment (martin-g) [inline wicket-util/src/main/java/org/apache/wicket/util/string/StringValue.java]: `xxx`

File: wicket-jmx/src/main/java/org/apache/wicket/jmx/wrapper/RequestCycleSettings.java (hunk 4 above)
Comment (bitstorm) [reply wicket-jmx/src/main/java/org/apache/wicket/jmx/wrapper/RequestCycleSettings.java]: @martin-g  I think you've lost me here :-). The old version of this method took in input a String and used a custom 'human-friendly' format to build a duration. For example something like '1.5 days' would have been converted into 36 hours. I've simply removed this textual format in order to use directly a standard Duration class.

Comment (martin-g) [reply wicket-jmx/src/main/java/org/apache/wicket/jmx/wrapper/RequestCycleSettings.java]: JMX works with a text-based protocol, like HTTP.
E.g. a user opens a UI and enters the values for the exposed settings. 
If the type of the property is a String or a primitive then all is fine (e.g. JMX will just use `Integer.parseInt(input)`). 
But if the of the property is of some complex type then someone has to help JMX to convert the String input to the target type.
I doubt there is a default converter for `java.time.Duration`. 

You can test it by starting a Wicket app and then connect to the JVM via JConsole or similar tool (e.g. wicketstuff-jmx-panel) and try to set a new value for `RequestCycleSettings#timeout`

File: wicket-core/src/test/java/org/apache/wicket/ajax/AjaxRequestHandlerTest.java (hunk 2 above)
Comment (bitstorm) [reply wicket-core/src/test/java/org/apache/wicket/ajax/AjaxRequestHandlerTest.java]: It seems the right RFC for Date formats is RFC7231:
https://tools.ietf.org/html/rfc7231#section-7.1.1.1

But I admit to be a little confused, especially in respect of what RFC1123 actually says about Date formats.

**Comment by bitstorm** [comment]
@martin-g I would proceed merging this PR If there are no other change requests.

**Comment by martin-g** [comment]
Go ahead!

**Pull request merged**